abr_control/controllers/path_planners/second_order.py

```diff
@@ -61,9 +71,21 @@ def step(self, state, target_pos, dt=0.001):
         n_states = int(len(state)/2)
         y = state[:n_states]
         dy = state[n_states:]
-
+        dist = np.linalg.norm(y - target_pos)
         w = self.w
-        if np.linalg.norm(y - target_pos) < self.threshold:
+
+        # uncomment for toolbot demo
+        # TODO: incorporate this hack properly
```

Comment:
this is the hack I added to vary `w` based on a linear function of distance to target. It's what I used in the demo to get the movement faster for short reaches without making long reaches unstable. I'm not sure how we could incorporate this in a more polished way, or if we want to move to a different path planner for the demo

Reply:
i like the latter, putting this either as a class on the demo branch, or in dewolf2018